picoGPT/demo.py

```
@@ -0,0 +1,120 @@
+        elif resp_count == 1:
+            item = resp[0]  # 取第一个即可
+            err_no = int(item.get("error_start_line_number"))
+            if not cls.valid_line_md5(task, record, err_no):
+                return None
+
+            # 返回回来的err_no已经是绝对行号,然后获取行号的上下三行存进context以便提 issue
+            line_context = cls.get_line_context(record, task.new_path, err_no)
+            # 将review出问题的行和其上下三行更新到数据库以便之后提问题
+            cls.update_by_id(mid, err_line_no=err_no, line_context=line_context)
+
+            source_code = utils.get_file_content(record, task.new_path)
```

Ruff (0.8.2)

Undefined name `utils`

(F821)